src/Commands.hs

```diff
@@ -0,0 +1,166 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Commands
+  ( runCommand
+  ) where
+
+import Control.Monad (forM_, when)
+import Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+import Data.Text (Text)
+import qualified Data.Text as T
+import qualified Data.Text.IO as TIO
+import System.Console.ANSI
+import System.Directory (makeAbsolute, doesFileExist, createDirectoryIfMissing)
+import System.FilePath (takeDirectory, (</>))
+import Types
+import Config
+import CSV
+import Git
+import Analysis
+import StackYaml
+import ColorOption
+import License (licenseText)
+import qualified XDG
+
+-- | Run a command
+runCommand :: Options -> IO ()
+runCommand opts = do
+  useColor <- shouldUseColor (optColor opts)
+  case optCommand opts of
+    Version -> printVersion
+    NumericVersion -> putStrLn appVersion
+    PrintLicense -> putStrLn licenseText
+    Help -> printHelp
+    cmd -> runEssentialCommand useColor cmd
+
+-- | Run an essential command (requires repo setup)
+runEssentialCommand :: Bool -> Command -> IO ()
+runEssentialCommand useColor cmd = do
+  repoPath <- getRepoPath
+  ensureRepo repoPath
+  
+  -- Ensure CSV files exist
+  stateDir <- XDG.getStateDir
+  csvExists <- doesFileExist (stateDir </> "ghc.csv")
+  when (not csvExists) $ generateCSVs repoPath
+  
+  case cmd of
+    DryRun -> runDryRun useColor
+    Bump -> runBump
+    Update -> runUpdate repoPath
+    Info -> runInfo repoPath
+    Config configCmd -> runConfig configCmd
+    _ -> return ()
+
+-- | Print version information
+printVersion :: IO ()
+printVersion = do
+  putStrLn $ appName ++ " version " ++ appVersion
+  putStrLn copyright
+
+-- | Print help
+printHelp :: IO ()
+printHelp = do
+  printVersion
+  putStrLn ""
+  putStrLn "A tool to bump snapshots (resolvers) in stack*.yaml files."
+  putStrLn ""
```

Comment:
The rest of the text should be generated from the command and options data defined in `Options.hs` rather than put verbatim here.